### PR TITLE
Covert config private key to buffer before deriving addresses.

### PR DESCRIPTION
**Pull request by swaylock**
resolves gh-1885

- [ ] Because this PR includes a **documentation change**, its branch was created from the `website` branch, and this PR uses the `website` branch as its base branch.
- [ ] Because this PR includes a **bug fix**, relevant tests have been included.
- [ ] Because this PR includes a **new feature**, the change was previously discussed on an Issue or with someone from the team.
- [x] I didn't do anything of this.

---

Here's our crack at fixing the reported error.

### CODE REVIEW

**Comment by swaylock** [comment]
I need to review this more. Something doesn't look right...

File: packages/hardhat-truffle5/src/index.ts [hunk 1]
```diff
@@ -136,9 +136,10 @@ extendEnvironment((env) => {
 
         accounts = normalizeHardhatNetworkAccountsConfig(
           netConfig.accounts
-        ).map((acc) =>
-          toChecksumAddress(bufferToHex(privateToAddress(acc.privateKey)))
-        );
+        ).map((acc) => {
+          const buffer = Buffer.alloc(32, acc.privateKey);
```
Comment (alcuadrado) [inline packages/hardhat-truffle5/src/index.ts:140]: I think you want to use `ethereumjs-utils`' `toBuffer` instead

**Review by alcuadrado — COMMENTED** [review]
Thanks for submitting this PR!

I left a minor comment, because I think this isn't working.

Would you mind adding a test?

**Comment by alcuadrado** [comment]
Oh, you close it while I was writing my review

File: packages/hardhat-truffle5/src/index.ts (hunk 1 above)
Comment (swaylock) [reply packages/hardhat-truffle5/src/index.ts:140]: Eek. sorry. `toBuffer` was the fix. Will re-open the PR and try to push to it.

**Comment by swaylock** [comment]
Hi @alcuadrado if you can point me in the right direction, I'd be glad to write the test.

**Comment by alcuadrado** [comment]
I added tests and a truffle4 version. Thanks @swaylock for starting this!

I had to create a new PR for this. So I'm closing this one.

**Pull request closed without merging**